src/error.rs

```diff
@@ -287,25 +285,51 @@ impl From<ShapeError> for Error {
     }
 }
 
-pub fn is_err_code<T>(value: T) -> bool
-where
-    T: Integer + Zero + Bounded + Copy,
-{
-    if T::min_value() < T::zero() {
-        value < T::zero()
-    } else {
-        value == T::zero()
+pub fn h5check<T: H5ErrorCode>(value: T) -> Result<T> {
+    H5ErrorCode::h5check(value)
+}
+
+#[allow(unused)]
+pub fn is_err_code<T: H5ErrorCode>(value: T) -> bool {
+    H5ErrorCode::is_err_code(value)
+}
+
+pub trait H5ErrorCode: Copy {
+    fn is_err_code(value: Self) -> bool;
+    fn h5check(value: Self) -> Result<Self> {
+        if Self::is_err_code(value) {
+            Error::query().map_or_else(|| Ok(value), Err)
+        } else {
+            Ok(value)
+        }
     }
 }
 
-pub fn h5check<T>(value: T) -> Result<T>
-where
-    T: Integer + Zero + Bounded + Copy,
-{
-    if is_err_code(value) {
-        Error::query().map_or_else(|| Ok(value), Err)
-    } else {
-        Ok(value)
+impl H5ErrorCode for hsize_t {
+    fn is_err_code(value: Self) -> bool {
+        value == 0
+    }
+}
+impl H5ErrorCode for herr_t {
+    fn is_err_code(value: Self) -> bool {
+        value < 0
+    }
+}
+#[cfg(hdf5_1_10_0)]
+impl H5ErrorCode for hid_t {
+    fn is_err_code(value: Self) -> bool {
+        value < 0
+    }
+}
+#[cfg(not(hdf5_1_10_0))]
+impl H5ErrorCode for hdf5_sys::h5::hssize_t {
```

Comment:
`use hdf5_sys::h5::hssize_t` above?

Reply:
Would require a `cfg` on that line too

Reply:
That's ok, I think, to make it inline with other modules where there's tons of cfg imports